### PR TITLE
how-to: add doc for upgrading TiDB using TiUP

### PR DESCRIPTION
**Pull request by kissmydb**
### What is changed, added or deleted? (Required)

document how to upgrade TiDB using TiUP

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -67,6 +67,7 @@
     - [使用 Ansible 扩容缩容](/how-to/scale/with-ansible.md)
   + 升级
     - [升级至最新开发版](/how-to/upgrade/from-previous-version.md)
```
Comment (yikeke) [inline TOC.md:69]:
```suggestion
    - [使用 TiDB Ansible 升级](/how-to/upgrade/from-previous-version.md)
```
为了与新加进来的“使用 TiUP 升级”文档逻辑上保持一致，这个文档标题是不是应该也相应地改一下？（甚至文件名、文档第一段的内容等都建议改一下） @kissmydb

File: how-to/upgrade/using-tiup.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+- 不支持在升级后回退至 3.0 版本。
+- 3.0 之前的版本，需要先通过 TiDB Ansible 升级到 3.0 版本，然后按照本文档的说明，使用 TiUP 将 TiDB Ansible 配置导入，再升级到 4.0 版本。
+- TiDB Ansible 配置导入到 TiUP 中管理后，不能再通过 TiDB Ansible 对集群进行操作，否则可能因元信息不一致造成冲突。
+- 对于满足以下情况之一的 TiDB Ansible 部署的集群，暂不支持导入：
+    - 启用了 `TLS` 加密功能的集群
+    - 纯 KV 集群（没有 TiDB 实例的集群）
+    - 启用了 `Kafka` 的集群
+    - 启用了 `Spark` 的集群
+    - 启用了 `Lightning` / `Importer` 的集群
+    - 仍使用老版本 `'push'` 的方式收集监控指标（从 3.0 默认为 `'pull'` 模式，如果没有特意调整过则可以支持）仍在使用 PushGateway 收集监控指标的集群
```
Comment (yikeke) [inline how-to/upgrade/using-tiup.md:23]: 仍使用老版本 `'push'` 的方式收集监控指标 和 仍在使用 PushGateway 收集监控指标 重复了？

**Comment by superlzs0476** [comment]
LGTM

File: how-to/upgrade/using-tiup.md (hunk 2 above)
Comment (TomShawn) [reply how-to/upgrade/using-tiup.md:23]: @kissmydb PTAL

**Comment by TomShawn** [comment]
> @TomShawn Please help remove numbers in headings. Thanks!

Done. PTAL.

File: how-to/upgrade/using-tiup.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
```
Comment (lilin90) [inline how-to/upgrade/using-tiup.md:23]:
```suggestion
    - 仍使用老版本 `'push'` 的方式收集监控指标（从 3.0 默认为 `'pull'` 模式，如果没有特意调整过则可以支持）
    - 仍在使用 PushGateway 收集监控指标的集群
```

File: TOC.md [hunk 4]
```diff
@@ -67,7 +67,8 @@
   + 扩容缩容
     - [使用 Ansible 扩容缩容](/how-to/scale/with-ansible.md)
   + 升级
-    - [升级至最新开发版](/how-to/upgrade/from-previous-version.md)
+    - [使用 TiDB Ansible 升级](/how-to/upgrade/from-previous-version.md)
+    - [使用 TiUP 升级](/how-to/upgrade/using-tiup.md)
```
Comment (kissmydb) [inline TOC.md:71]:
```suggestion
    - [使用 TiUP 升级（推荐）](/how-to/upgrade/using-tiup.md)
```

File: TOC.md [hunk 5]
```diff
@@ -67,7 +67,8 @@
   + 扩容缩容
     - [使用 Ansible 扩容缩容](/how-to/scale/with-ansible.md)
   + 升级
-    - [升级至最新开发版](/how-to/upgrade/from-previous-version.md)
+    - [使用 TiDB Ansible 升级](/how-to/upgrade/from-previous-version.md)
```
Comment (kissmydb) [inline TOC.md:70]: 这个放到 TiUP 后面吧，把 tiup 放前面，并（推荐）

File: how-to/upgrade/using-tiup.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+- 3.0 之前的版本，需要先通过 TiDB Ansible 升级到 3.0 版本，然后按照本文档的说明，使用 TiUP 将 TiDB Ansible 配置导入，再升级到 4.0 版本。
+- TiDB Ansible 配置导入到 TiUP 中管理后，不能再通过 TiDB Ansible 对集群进行操作，否则可能因元信息不一致造成冲突。
+- 对于满足以下情况之一的 TiDB Ansible 部署的集群，暂不支持导入：
+    - 启用了 `TLS` 加密功能的集群
+    - 纯 KV 集群（没有 TiDB 实例的集群）
+    - 启用了 `Kafka` 的集群
+    - 启用了 `Spark` 的集群
+    - 启用了 `Lightning` / `Importer` 的集群
+    - 仍使用老版本 `'push'` 的方式收集监控指标（从 3.0 默认为 `'pull'` 模式，如果没有特意调整过则可以支持）
+    - 仍在使用 PushGateway 收集监控指标的集群
```
Comment (kissmydb) [inline how-to/upgrade/using-tiup.md:24]:
```suggestion
```

**Comment by TomShawn** [comment]
LGTM

**Pull request merged**